app/controllers/admin/deploy_groups_controller.rb

```diff
@@ -82,18 +82,43 @@ def create_all_stages_preview
   # No more than one stage, per project, per deploy_group
   # Note: you can call this multiple times, and it will create missing stages, but no redundant stages.
   def create_all_stages
-    _, missing_stages = stages_for_creation
-    missing_stages.each do |template_stage|
-      create_stage_with_group(template_stage)
+    self.class.create_all_stages(deploy_group)
+
+    redirect_to [:admin, deploy_group]
+  end
+
+  def merge_all_stages
+    preexisting_stages, = stages_for_creation
+    template_stages = deploy_group.environment.template_stages.all
+
+    preexisting_stages.each do |stage|
```

Comment:
this also includes stages that have multiple deploy groups ... that might be an issue that should be addresses in stages_for_creation